src/EventListener/CommandSubscriber.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Chrisguitarguy\RequestId\EventListener;
+
+use Symfony\Component\Console\ConsoleEvents;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Chrisguitarguy\RequestId\RequestIdGenerator;
+use Chrisguitarguy\RequestId\RequestIdStorage;
+
+/**
+ * Set up request id for command.
+ *
+ * @since   1.0
+ */
+final class CommandSubscriber implements EventSubscriberInterface
+{
+    private RequestIdStorage $requestIdStorage;
+    private RequestIdGenerator $generator;
+
+    public function __construct(
+        RequestIdStorage $requestIdStorage,
+        RequestIdGenerator $generator
+    )
+    {
+        $this->requestIdStorage = $requestIdStorage;
+        $this->generator = $generator;
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            ConsoleEvents::COMMAND => [
+                ['onCommand', 999],
+            ],
+            ConsoleEvents::TERMINATE => [
+                ['onTerminate', -999],
+            ],
+        ];
+    }
+
+    public function onCommand(): void
+    {
+        $this->requestIdStorage->setRequestId("CLI{$this->generator->generate()}");
```

Comment:
is it make any sense for long running operations? i mean, how it is supposed to help if you are consuming many business operation in one command like symfony messenger:consume or some type of corne

how about implementing messenger middleware that add unique request id to each messages? or event pass form dispaching process and try to read from message after all?